src/bits.c

```diff
@@ -1147,22 +1147,32 @@ bit_read_CRC(Bit_Chain * dat)
 /** Read and check CRC-number.
  */
 int
-bit_check_CRC(Bit_Chain * dat, long unsigned int start_address,
-              uint16_t seed)
+bit_check_CRC(Bit_Chain * dat, long unsigned int start_address, 
+              long unsigned int size, uint16_t seed)
 {
   unsigned int calculated;
   unsigned int read;
   unsigned char res[2];
+  long unsigned int old_byte;
+  long unsigned int old_bit;
 
-  if (dat->bit > 0)
-    dat->byte++;
-  dat->bit = 0;
+  if (start_address + size >= dat->size)
+    {
+      LOG_WARN("bit_check_CRC out of range");
+      return 0;
+    }
 
-  calculated = bit_calc_CRC(seed, &(dat->chain[start_address]),
-                            dat->byte - start_address);
+  old_byte = dat->byte;
+  old_bit = dat->bit;
 
+  calculated = bit_calc_CRC(seed, &(dat->chain[start_address]), size);
```

Comment:
No. This is wrong for unaligned bits. We need to align dat first, as in `if (dat->bit > 0) dat->byte++; dat->bit = 0;`

Reply:
Why we need do this, if we have `start_address` parametr? Current `dat` bit and byte positions are not considered.

Reply:
start_address is only the byte offset, with bit still 0.
What we need before that, is to do the alignment. You removed the alignment code.